pkg/secrethub/credentials/creators.go

```diff
@@ -31,13 +34,36 @@ func CreateKey() *KeyCreator {
 }
 
 // Create generates a new key and stores it in the KeyCreator.
-func (c *KeyCreator) Create() (Verifier, Encrypter, map[string]string, error) {
+func (kc *KeyCreator) Create() error {
 	key, err := GenerateRSACredential(crypto.RSAKeyLength)
 	if err != nil {
-		return nil, nil, nil, err
+		return err
 	}
-	c.key = key
-	return c.key, c.key, map[string]string{}, nil
+	kc.key = key
+	return nil
+}
+
+func (kc *KeyCreator) Metadata() map[string]string {
+	return map[string]string{}
+}
+
+type awsCreator struct {
+	kmsKeyID string
+	roleARN  string
+	awsCfg   []*awssdk.Config
+
+	verifier  Verifier
+	encrypter Encrypter
+}
+
+func (ac *awsCreator) Create() error {
+	creator, err := aws.NewCredentialCreator(ac.kmsKeyID, ac.roleARN, ac.awsCfg...)
+	if err != nil {
+		return err
+	}
+	ac.encrypter = creator
+	ac.verifier = creator
```

Comment:
why does this have two fields that are the same? Can we then have one field that are returned for both the `Verifier` and `Authenticator` functions?

Reply:
:check: